source/imaer-gml/src/main/java/nl/overheid/aerius/gml/base/source/ship/v31/GML2InlandMooring.java

```diff
@@ -131,12 +143,24 @@ private int determinePercentageLaden(final int shipsExisting, final int percenta
     return (percentageLadenExisting * shipsExisting + percentageLadenToAdd * shipsToAdd) / (shipsExisting + shipsToAdd);
   }
 
-  private InlandShippingEmissionSource createRoute(final IsGmlProperty<IsGmlInlandWaterway> waterwayProperty) {
+  protected InlandShippingEmissionSource createRoute(final IsGmlProperty<IsGmlInlandWaterway> waterwayProperty, final String sourceLabel,
+      final AtomicInteger routeIndexTracker) {
     final InlandShippingEmissionSource mooringRoute = new InlandShippingEmissionSource();
     final InlandWaterway waterway = GML2InlandUtil.toInlandWaterway(waterwayProperty);
     mooringRoute.setSectorId(DEFAULT_ROUTE_SECTOR_ID);
     mooringRoute.setWaterway(waterway);
+    mooringRoute.setLabel(constructLabel(sourceLabel, routeIndexTracker));
     return mooringRoute;
   }
 
+  protected String constructLabel(final String sourceLabel, final AtomicInteger routeIndexTracker) {
+    // Routes did not have a name in old GML, so come up with our own label
+    // Might not be completely i18n, but works for both NL and EN at least.
+    final int currentIndex = routeIndexTracker.getAndIncrement();
+    final String routeDescription = "Route " + currentIndex;
+    return Stream.of(sourceLabel, routeDescription)
+        .filter(x -> x != null && !x.isBlank())
+        .collect(Collectors.joining("; "));
```

Comment:
Looks like a somewhat complex way of writing. But is a personal preference:
```suggestion
    return (sourceLabel == null || sourceLabel.isBlank() ? "" : (sourceLabel + "; ")) + routeDescription;
```

Reply:
yeah, used that construction because I've used it in other places as well. Instead of changing it to your suggestion, moved it to base class so it can be re-used.